corokafka/impl/corokafka_consumer_manager_impl.cpp

```diff
@@ -578,8 +582,7 @@ cppkafka::Error ConsumerManagerImpl::commitImpl(ConsumerTopicEntry& entry,
         if (entry._committer->get_consumer().get_configuration().get_offset_commit_callback() && (opaque != nullptr)) {
             entry._offsets.insert(headPartition, opaque);
         }
-        if ((entry._autoOffsetPersistStrategy == OffsetPersistStrategy::Commit) ||
-            (execMode == ExecMode::Sync)) {
+        if (entry._autoOffsetPersistStrategy == OffsetPersistStrategy::Commit) {
```

Comment:
Let's update CONFIGURATION.md to point out that if you've configured internal.consumer.offset.persist.strategy=store, the execMod is not checked.